tests/integration-tests/src/commonTest/kotlin/test/WatcherTest.kt

```diff
@@ -330,14 +331,17 @@ class WatcherTest {
     val channel = Channel<EpisodeHeroNameQuery.Data?>()
 
     // This will initially miss as the cache should be empty
-    val job = launch(start = CoroutineStart.UNDISPATCHED) {
+    val job = launch {
       apolloClient.query(query)
           .watch(null)
           .collect {
             channel.send(it.data)
           }
     }
 
+    // Because subscribe is called from a background thread, give some time to be effective
+    delay(500)
```

Comment:
That's an interesting one. If we want a guarantee that `.watch()` subscribes synchronously, we probably need to go back to "interceptors do their own dispatcher switch". I want to measure how costly the thread switch is.